### DISCUSSION TITLE
Can I export my Graph as PDF or Image?

### DISCUSSION THREAD

**Discussion by HalilFocic** [maxGraph/maxGraph#604] [Q&A]
Hi, is it possible to somehow extract my graph as PDF or as Image? I would love to send it to my backend as file or base64 so it can be stored and presented in modal on mobile devices.

**Comment by tbouffard** [comment]
Hi AFAIK, this is not possible to do this.

It is possible to do SVG export, I do it with mxGraph and this should work as well in maxGraph. Then, if you send the SVG to your backend you could generate the PDF backend side. Or doing the SVG to PDF conversion in the browser, but this is probably not a performant way to achieve your goal 😄 .
SVG export involves the use of `ImageExport`. 
There is an example here https://github.com/maxGraph/maxGraph/blob/v0.14.0/packages/html/stashed/grapheditor/www/js/Graph.js#L8188 but drawio probably provides a more up to date version of the code: 
- https://github.com/jgraph/drawio/blob/v24.7.17/src/main/webapp/plugins/svgdata.js#L9
- https://github.com/jgraph/drawio/blob/v24.7.17/src/main/webapp/js/grapheditor/Graph.js#L11688

Here is the code I used (adapted from the drawio code): https://github.com/process-analytics/bpmn-visualization-js/blob/3be35d4289054f04bc7343c37613bb5a45f70a41/dev/ts/component/SvgExporter.ts#L33

In draw.io (and perhaps in the former mxGraph grapheditor examples) it is also possible to generate PDF. But this is done server side AFAIK.

[UPDATE] See also a past answer I wrote about this topic: https://github.com/maxGraph/maxGraph/discussions/234#discussioncomment-6848935

**Reply by HalilFocic** [reply]
Hi @tbouffard .
Thank you so much. Even svg is okay, i need to somehow present static graph on mobile so I think svg will work.

Thanks again

**Reply by tbouffard** [reply]
You are welcome. 😄 

Another thought if you don't use `htmlLabels`.
In this case, this is even simpler: you can get the SVG content from the DOM.

When using `htmlLabels`, maxGraph put the label in foreignObject and the position of the text is highly dependant of the rest of the page. That's why in this case, we have to use the ImageExport and some tricks to correctly position the labels in the exported SVG.
